ui/src/utils/cache.ts

```diff
@@ -0,0 +1,116 @@
+// cache invalidation source-of-truth
```

Comment:
I like the concept of cache management utils, but not sure if one big object is a good idea.

WDYT about utils hook inside a feature folder?

Reply:
Definitely I can add

Reply:
It's better, but we don't need those hooks in the utils (common) folder.

`analytics-template` we can move to `/features/project/analysis-templates/utils/query-data-cache.ts` or something like this. WDYT?